src/structures/gdt.rs

```diff
@@ -306,17 +307,62 @@ impl Descriptor {
     /// Creates a TSS system descriptor for the given TSS.
     #[inline]
     pub fn tss_segment(tss: &'static TaskStateSegment) -> Descriptor {
+        // SAFETY: if iomap_size is zero, there are no requirements to uphold.
+        unsafe { Self::tss_segment_raw(tss, 0) }
+    }
+
+    /// Creates a TSS system descriptor for the given TSS, setting up the IO permissions bitmap.
+    pub fn tss_segment_with_iomap(
+        tss: &'static TaskStateSegment,
+        iomap: &'static [u8],
+    ) -> Result<Descriptor, InvalidIoMap> {
+        if iomap.len() > 8193 {
+            return Err(InvalidIoMap::TooLong { len: iomap.len() });
+        }
+
+        let base = iomap.as_ptr() as usize - tss as *const _ as usize;
+        if base > 0xdfff {
+            return Err(InvalidIoMap::TooFarFromTss { distance: base });
+        }
+
+        let last_byte = *iomap.last().unwrap_or(&0xff);
```

Comment:
Any thoughts on this? Should it be valid for a zero-length slice to be passed in?

Reply:
I don't have a strong opinion on this, so whatever you prefer.